From 23d198109f3bfd8a21c6f74e178a46d924256b7a Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Thu, 8 Dec 2016 20:12:49 +0800
Subject: [PATCH 1328/5242] MLK-13579-2 ARM: imx: update suspend ocram
 location on i.mx7ulp

commit  7214dc04c4e0d81b3e4ccf1efbecb3c713e3b0a9 from
https://source.codeaurora.org/external/imx/linux-imx.git

Adjust i.MX7ULP suspend ocram location from
SRAM_U to SRAM_L, occupy last 16K.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/suspend-imx7ulp.S |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-imx/suspend-imx7ulp.S b/arch/arm/mach-imx/suspend-imx7ulp.S
index c24b8cd..40a3ee3 100644
--- a/arch/arm/mach-imx/suspend-imx7ulp.S
+++ b/arch/arm/mach-imx/suspend-imx7ulp.S
@@ -58,6 +58,7 @@
 #define PM_INFO_MMDC_NUM_OFFSET			0x198
 #define PM_INFO_MMDC_VAL_OFFSET			0x19c
 
+#define DGO_CTRL0	0x50
 #define DGO_GPR3	0x60
 #define DGO_GPR4	0x64
 
@@ -446,7 +447,8 @@ resume:
 	orr	r7, r7, #(1 << 14)
 	str	r7, [r6, #PMC1_CTRL]
 
-	ldr	r0, =0x20008000
+	ldr	r6, =MX7ULP_SIM_BASE_ADDR
+	ldr	r0, [r6, #DGO_GPR4]
 	/* get physical resume address from pm_info. */
 	ldr	lr, [r0, #PM_INFO_RESUME_ADDR_OFFSET]
 
-- 
1.7.9.5

